sycl/include/CL/sycl/types.hpp

```diff
@@ -278,8 +275,17 @@ convertImpl(T Value) {
     // Round to nearest even is default rounding mode for floating-point types
   case rounding_mode::automatic:
     // Round to nearest even.
-  case rounding_mode::rte:
-    return std::round(Value);
+  case rounding_mode::rte: {
+    int OldRoundingDirection = std::fegetround();
+    int Err = std::fesetround(FE_TONEAREST);
```

Comment:
C++ reference says we have to set pragma FENV_ACCESS to safely use this.
http://www.cplusplus.com/reference/cfenv/fesetround/

Reply:
Thank you for noticing this. 
I added this statement.